src/plotting.jl

```diff
@@ -0,0 +1,413 @@
+
+function series!(ax::Axis, curves::XYVector; kw...)
+    series!(ax, [curves]; kw...)
+end
+
+function series!(ax::Axis, curves::AbstractVector{<: XYVector}; labels=nothing, linewidth=2, scatter=nothing, color=nothing)
+    colors = to_colormap(:lighttest, 8)
+    for (i, (x, y)) in enumerate(curves)
+        x = replace_missings(x)
+        y = replace_missings(y)
+        label = isnothing(labels) ? "" : labels[i]
+        series_color = isnothing(color) ? colors[i] : color
+        if !isnothing(scatter)
+            scatterlines!(ax, x, y; linewidth=linewidth, color=series_color, markercolor=series_color,
+                          label=label, scatter...)
+        else
+            lines!(ax, x, y; linewidth=linewidth, color=series_color, label=label)
+        end
+    end
+end
+
+function plot_pr_curves!(subfig::FigurePosition, per_class_pr_curves::SeriesCurves,
+                         class_labels::Union{Nothing, AbstractVector{String}}; legend=:lt, title="PR curves",
+                         xlabel="True positive rate", ylabel="Precision",
+                         linewidth=2, scatter=nothing, color=nothing)
+
+    ax = Axis(subfig;
+        title=title,
+        xlabel=xlabel, ylabel=ylabel,
+        xticks=0:0.2:1, yticks=0:0.2:1)
+
+    hidedecorations!(ax, label = false, ticklabels = false, grid=false)
+    xlims!(ax, 0, 1)
+    ylims!(ax, 0, 1)
+    series!(ax, per_class_pr_curves; labels=class_labels, linewidth=linewidth, scatter=scatter, color=color)
+    if !isnothing(legend)
+        axislegend(ax; position=legend)
+    end
+    return ax
+end
+
+function plot_prg_curves!(subfig::FigurePosition, per_class_prg_curves::SeriesCurves,
+                          per_class_prg_aucs::NumberVector,
+                          class_labels::AbstractVector{<: String};
+                          legend=:lt,
+                          title="PR-Gain curves",
+                          xlabel="True positive rate gain",
+                          ylabel="Precision gain")
+
+    auc_labels = [@sprintf("%s (AUC F1: %.3f)", class, per_class_prg_aucs[i])
+                  for (i, class) in enumerate(class_labels)]
+    return plot_pr_curves!(subfig, per_class_prg_curves, auc_labels; legend=legend, title=title,
+                           xlabel=xlabel, ylabel=ylabel)
+end
+
+function plot_roc_curves!(subfig::FigurePosition, per_class_roc_curves::SeriesCurves,
+                          per_class_roc_aucs::NumberVector,
+                          class_labels::AbstractVector{<: String};
+                          legend=:rb,
+                          title="ROC curves",
+                          xlabel="False positive rate",
+                          ylabel="True positive rate")
+
+    auc_labels = [@sprintf("%s (AUC: %.3f)", class, per_class_roc_aucs[i])
+                  for (i, class) in enumerate(class_labels)]
+
+    return plot_pr_curves!(subfig, per_class_roc_curves, auc_labels; legend=legend, title=title,
```

Comment:
It feels a little weird for this to call `plot_pr_curves!` under the hood; maybe indicates that `plot_pr_curves!(...)` should be renamed to something more generic? e.g. a `plot_curves!(...)` function that both `plot_roc_curves!` and `plot_pr_curves!` call.

Reply:
...now seeing that it also gets called by `plot_reliability_calibration_curves!(...)` below, and possibly others, so I'm even more in favor of a generic name for the shared functionality.

Reply:
Yes, thanks for the reminder...That was a stopgap solution to try out if I can use the same function for the others, and then never came up with a good name for it. Lets rename it and make `plot_pr_curves` call it!